### PR TITLE
Integration test for Foreign Key Action Migration Feature

### PR DESCRIPTION
**Pull request by aasthabharill**

### CODE REVIEW

File: testing/postgres/integration_test.go [hunk 1]
```diff
@@ -328,6 +376,43 @@ func checkArrays(ctx context.Context, t *testing.T, client *spanner.Client) {
...
+	}
+	defer client.Close()
+
+	// Verifying that the row to be deleted exists in parent - otherwise testing would be incorrect
+	stmt := spanner.Statement{SQL: `SELECT * FROM products WHERE productid = '1YMWWN1N4O'`}
+	iter := client.Single().Query(ctx, stmt)
+	defer iter.Stop()
+	row, _ := iter.Next()
+
+	assert.NotNil(t, row, "No row exists with given key in 'products'")
```
Comment (shreyakhajanchi) [inline testing/postgres/integration_test.go:392]: Add the key in the error msg

File: testing/postgres/integration_test.go [hunk 2]
```diff
@@ -328,6 +376,43 @@ func checkArrays(ctx context.Context, t *testing.T, client *spanner.Client) {
...
+
+	// Deleting row from parent table in Spanner DB
+	mutation := spanner.Delete("products", spanner.Key{"1YMWWN1N4O"})
+	_, err = client.Apply(ctx, []*spanner.Mutation{mutation})
+	// check iska behaviour
```
Comment (shreyakhajanchi) [inline testing/postgres/integration_test.go:397]: Delete this comment

File: testing/postgres/integration_test.go [hunk 3]
```diff
@@ -328,6 +376,43 @@ func checkArrays(ctx context.Context, t *testing.T, client *spanner.Client) {
...
+	if err != nil {
+		t.Fatalf("Failed to delete row: %v", err)
+	}
+
+	// Testing ON DELETE NO ACTION - row shouldn't have been deleted in parent and child
+	stmt = spanner.Statement{SQL: `SELECT * FROM products WHERE productid = '1YMWWN1N4O'`}
```
Comment (shreyakhajanchi) [inline testing/postgres/integration_test.go:403]: why should the row exist in product table after the delete query?

File: testing/mysql/integration_test.go [hunk 4]
```diff
@@ -283,6 +319,35 @@ func checkJson(ctx context.Context, t *testing.T, client *spanner.Client, dbURI
...
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	// Verifying that the row to be deleted exists in child - otherwise test will incorrectly pass
+	stmt := spanner.Statement{SQL: `SELECT * FROM cart WHERE product_id = "zxi-631"`}
+	iter := client.Single().Query(ctx, stmt)
+	defer iter.Stop()
+	_, err = iter.Next()
+	assert.Equal(t, nil, err, "Expected rows with product_id \"zxi-631\" in 'cart'")
```
Comment (shreyakhajanchi) [inline testing/mysql/integration_test.go:334]: Store row = iter.Next() and assertNotNil

File: testing/mysql/integration_test.go [hunk 5]
```diff
@@ -218,6 +216,44 @@ func TestIntegration_MySQLDUMP_SchemaAndDataSubcommand(t *testing.T) {
...
+	t.Parallel()
+
+	tmpdir := prepareIntegrationTest(t)
+	defer os.RemoveAll(tmpdir)
+
+	dbName := "mysql-foreignkey-actions"
+	dbURI := fmt.Sprintf("projects/%s/instances/%s/databases/%s", projectID, instanceID, dbName)
+	filePrefix := filepath.Join(tmpdir, dbName)
+
+	host, user, srcDb, password := os.Getenv("MYSQLHOST"), os.Getenv("MYSQLUSER"), "test_foreign_key_action_data", os.Getenv("MYSQLPWD")
```
Comment (shreyakhajanchi) [inline testing/mysql/integration_test.go:230]: better to store this DB name in the env variables too

**Review by shreyakhajanchi — APPROVED** [review]
LGTM

**Review by Deep1998 — APPROVED** [review]
LGTM

**Pull request merged**